routers/web/repo/setting.go

```diff
@@ -349,6 +356,16 @@ func SettingsPost(ctx *context.Context) {
 			RemoteName: fmt.Sprintf("remote_mirror_%s", remoteSuffix),
 			Interval:   interval,
 		}
+		if form.PushMirrorUsePublicKey {
+			publicKey, privateKey, err := crypto.GenerateEd25519Keypair()
```

Comment:
I have a question that why should Gitea generate the ssh key-pair for users, instead of allowing users upload pre-generated keys?

Reply:
[As noted here](https://github.com/go-gitea/gitea/issues/18159#issuecomment-1042112387) user don't have access to private keys. No need to worry they can leak it some way. And no need to transfer it over the internet(even though that is secure).

Reply:
I could add a option to allow it, but I prefer it doesn't, as I can already imagine users uploading their personal private keys.

Reply:
I know the concern. But I can imagine that some users with thousands mirrors would complain that why they have to setup thousand keys .....

Reply:
`>I could add a option to allow it,` I hope you mean with those users that they have their own instances?

Reply:
I am not sure and I have no preference. Just imagine a possible case.

If it's very sure that no customized keys should be supported, here needs more comments and UI prompts.

Reply:
I'm not per-se against adding a option, but the default should be that gitea generates them, safest option IMO.

Reply:
How to handle once user missed the public key?

Reply:
See screenshot, it is displayed permanently.